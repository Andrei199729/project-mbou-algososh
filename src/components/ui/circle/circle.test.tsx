import React from "react";
import renderer from "react-test-renderer";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Circle component", () => {
  it("без буквы", () => {
    const circle = renderer.create(<Circle />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`);
  });

  it("с буквами", () => {
    const circle = renderer.create(<Circle letter="word" />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      word
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`);
  });

  it("с head", () => {
    const circle = renderer.create(<Circle head="1" />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    1
  </div>
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`);
  });

  it("с react-элементом в head", () => {
    const circle = renderer.create(<Circle head={<Circle />} />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`);
  });

  it("с tail", () => {
    const circle = renderer.create(<Circle tail="tail" />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`);
  });

  it("с react-элементом в tail", () => {
    const circle = renderer.create(<Circle tail={<Circle />} />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`);
  });

  it("с index", () => {
    const circle = renderer.create(<Circle index={1} />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`);
  });

  it("с пропом isSmall ===  true", () => {
    const circle = renderer.create(<Circle isSmall={true} />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle small default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`);
  });

  it("в состоянии default", () => {
    const circle = renderer.create(<Circle state={ElementStates.Default} />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`);
  });

  it("в состоянии changing", () => {
    const circle = renderer.create(<Circle state={ElementStates.Changing} />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  changing"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`);
  });

  it("в состоянии modified", () => {
    const circle = renderer.create(<Circle state={ElementStates.Modified} />);
    expect(circle).toMatchSnapshot(`
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  modified"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`);
  });
});
