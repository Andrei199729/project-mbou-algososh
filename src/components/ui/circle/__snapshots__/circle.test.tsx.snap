// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component без буквы: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component в состоянии changing: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  changing"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  changing"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component в состоянии default: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component в состоянии modified: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  modified"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  modified"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component с head: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    1
  </div>
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-cy="head"
  >
    1
  </div>
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component с index: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component с react-элементом в head: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  >
    <div
      className="content "
      data-cy="circle-cont"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
      />
      <div
        className="circle  default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-cy="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
      />
    </div>
  </div>
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component с react-элементом в tail: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle  default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  >
    <div
      className="content "
      data-cy="circle-cont"
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
        data-cy="head"
      />
      <div
        className="circle  default"
        data-cy="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
        data-cy="index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-cy="tail"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component с tail: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-cy="tail"
  >
    tail
  </div>
</div>
`;

exports[`Circle component с буквами: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      word
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle  default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      word
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;

exports[`Circle component с пропом isSmall ===  true: 
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle small default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
 1`] = `
<div
  className="content "
  data-cy="circle-cont"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-cy="head"
  />
  <div
    className="circle small default"
    data-cy="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-cy="index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-cy="tail"
  />
</div>
`;
